packages/components/src/public/QueryModel/SearchBox.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { ChangeEvent, FC, FormEvent, memo, useCallback, useEffect, useState, useMemo } from 'react';
+import { ActionValue } from '../../internal/components/omnibox/actions/Action';
+
+interface Props {
+    actionValues: ActionValue[];
+    onSearch: (value: string) => void;
+}
+
+export const SearchBox: FC<Props> = memo(props => {
+    const { actionValues, onSearch } = props;
+    const [searchValue, setSearchValue] = useState('');
+    const appliedSearch = useMemo(
+        () => actionValues.find(actionValue => actionValue.action.keyword === 'search')?.value,
```

Comment:
We seem to do this filtering to find particular types of actions in a number of places.  I wonder if one or more utility methods would be called for.

Reply:
done